### PR TITLE
PV2-248: change period to delivery and collection

### PR DESCRIPTION
**Pull request by kuzinator**
this request includes a fix for https://skillsfundingagency.atlassian.net/browse/PV2-273

### CODE REVIEW

File: src/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsService/RequiredPaymentsService.cs [hunk 1]
```diff
@@ -45,7 +45,7 @@ public async Task<ApprenticeshipContractType2RequiredPaymentEvent> HandleAct2Pay
         {
             paymentLogger.LogVerbose($"Handling PaymentDue for {apprenticeshipKey}");
 
-            if (!await IsInitialised().ConfigureAwait(false))
+            //if (!await IsInitialised().ConfigureAwait(false))
```
Comment (davehowlett) [inline src/SFA.DAS.Payments.RequiredPayments.RequiredPaymentsService/RequiredPaymentsService.cs]: Do we not need this now? If not, can it be removed?

File: src/SFA.DAS.Payments.RequiredPayments.AcceptanceTests/Steps/RequiredPaymentsSteps.cs [hunk 2]
```diff
@@ -48,7 +49,7 @@ private ApprenticeshipContractTypePaymentDueEvent CreatePaymentDueEvent(Data.OnP
             payment.DeliveryPeriod = new CalendarPeriod(CollectionYear, paymentDue.DeliveryPeriod);
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.AcceptanceTests/Steps/RequiredPaymentsSteps.cs]: Not any more. Removed.

File: src/SFA.DAS.Payments.RequiredPayments.AcceptanceTests/Steps/RequiredPaymentsSteps.cs [hunk 3]
```diff
@@ -57,54 +58,54 @@ public void ThenTheRequiredPaymentsComponentWillGenerateTheFollowingContractType
         {
             var expectedPaymentsEvents = table
                 .CreateSet<OnProgrammePaymentDue>().ToArray();//TODO: fix to use a required payments model
-            WaitForIt(() => MatchRequiredPayment(expectedPaymentsEvents), "Failed to find all the required payment events");
+            WaitForIt(() => MatchRequiredPayment(expectedPaymentsEvents, null), "Failed to find all the required payment events");
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.AcceptanceTests/Steps/RequiredPaymentsSteps.cs]: This was done deliberately for each caller to make conscious decision on what to pass there.
Changed this to default null anyway as all callers are done.

File: src/SFA.DAS.Payments.RequiredPayments.AcceptanceTests/Steps/RequiredPaymentsSteps.cs [hunk 4]
```diff
@@ -57,54 +58,54 @@ public void ThenTheRequiredPaymentsComponentWillGenerateTheFollowingContractType
...
         }
 
-        private bool MatchUnexpectedRequiredPayment()
+        private bool MatchUnexpectedRequiredPayment(OnProgrammeEarningType? type)
         {
-            var result = PaymentsDue.Where(x => x.Type == OnProgrammeEarningType.Balancing).All(paymentDue =>
+            var result = PaymentsDue.Where(x => !type.HasValue || x.Type == type).ToList().All(paymentDue =>
                 !ApprenticeshipContractType2Handler.ReceivedEvents.Any(receivedEvent =>
                     paymentDue.Amount == receivedEvent.AmountDue
                     && TestSession.Learner.LearnRefNumber == receivedEvent?.Learner?.ReferenceNumber
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.AcceptanceTests/Steps/RequiredPaymentsSteps.cs]: done

File: src/SFA.DAS.Payments.RequiredPayments.AcceptanceTests/Steps/RequiredPaymentsSteps.cs [hunk 5]
```diff
@@ -123,9 +124,23 @@ private bool MatchUnexpectedRequiredPayment()
...
+            }
+            else
+            {
+                events = expectedPaymentsEvents;
+            }
+
+            var allFound = events.All(expectedEvent =>
                 ApprenticeshipContractType2Handler.ReceivedEvents.Any(receivedEvent =>
                     expectedEvent.Amount == receivedEvent.AmountDue
                     && TestSession.Learner.LearnRefNumber == receivedEvent?.Learner?.ReferenceNumber
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.AcceptanceTests/Steps/RequiredPaymentsSteps.cs]: done

File: src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Handlers/ApprenticeshipContractType2PaymentDueEventHanlderTest.cs [hunk 6]
```diff
@@ -205,6 +205,9 @@ public async Task TestPopulateHistory()
             _apprenticeshipKeyServiceMock.Setup(s => s.GeneratePaymentKey("2", "2", 3, It.IsAny<CalendarPeriod>())).Returns("1").Verifiable();
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Handlers/ApprenticeshipContractType2PaymentDueEventHanlderTest.cs]: done

File: src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Handlers/ApprenticeshipContractType2PaymentDueEventHanlderTest.cs [hunk 7]
```diff
@@ -205,6 +205,9 @@ public async Task TestPopulateHistory()
...
             _apprenticeshipKeyServiceMock.Setup(s => s.GeneratePaymentKey("3", "2", 3, It.IsAny<CalendarPeriod>())).Returns("2").Verifiable();
```
[the remark this answers is not in the document]
Comment (kuzinator) [reply src/SFA.DAS.Payments.RequiredPayments.Application.UnitTests/Application/Handlers/ApprenticeshipContractType2PaymentDueEventHanlderTest.cs]: done

**Pull request merged**